protocol/src/main/java/tw/waterball/survive/io/protocol/scanner/ModuleClassesScanner.java

```diff
@@ -0,0 +1,81 @@
+
+    }
+
+    public static Set<Class<?>> scanClassesByPackageName(String packageName) {
+        // 第一個class類的集合
+        Set<Class<?>> classes = new HashSet<>();
+        // 獲取包的名字並進行替換
+        String packageDirName = packageName.replace('.', '/');
+        try {
+            // 定義一個枚舉的集合併進行循環來處理這個目錄下的things
+            Enumeration<URL> dirs = Thread.currentThread().getContextClassLoader().getResources(packageDirName);
+            // 循環迭代下去
+            while (dirs.hasMoreElements()) {
+                // 獲取下一個元素
+                URL url = dirs.nextElement();
+                // 得到協議的名稱
+                String protocol = url.getProtocol();
+                // 如果是以文件的形式保存在服務器上
+                if ("file".equals(protocol)) {
+                    // 獲取包的物理路徑
+                    String filePath = URLDecoder.decode(url.getFile(), "UTF-8");
+                    // 以文件的方式掃描整個包下的文件並添加到集合中
+                    findClassesInPackageByFile(packageName, filePath, classes);
+                }
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return classes;
+    }
+
+    /**
+     * 以文件的形式來獲取包下的所有Class
+     *
+     * @param packageName
+     * @param packagePath
+     * @param classes
+     */
+    private static void findClassesInPackageByFile(String packageName, String packagePath, Set<Class<?>> classes) {
+        // 獲取此包的目錄建立一個File
+        File directory = new File(packagePath);
+        if (directory.exists() && directory.isDirectory()) {
+            // 如果存在就獲取包下的所有文件包括目錄
+            // 自定義過濾規則如果可以循環(包含子目錄) 或則是以.class結尾的文件(編譯好的java類文件)
+            File[] dirFiles = directory.listFiles();
+            // 循環所有文件
+            Arrays.stream(Objects.requireNonNull(dirFiles))
+                    .parallel()
+                    .forEach(file -> {
+                        // 如果是目錄則繼續掃描
+                        String fileName = file.getName();
+                        if (file.isDirectory()) {
+                            findClassesInPackageByFile(packageName + "." + fileName, file.getAbsolutePath(), classes);
+                        } else {
+                            try {
+                                // 如果是java類文件去掉後面的.class 只留下類名
+                                String className = fileName.substring(0, fileName.length() - 6);
+                                // 添加到集合中去
+                                // classes.add(Class.forName(packageName + '.' + className));
+                                // 經過回復同學的提醒，這裡用forName有一些不好，會觸發static方法，沒有使用classLoader的load乾淨
+                                String path = packageName + '.' + className;
+                                classes.add(Thread.currentThread().getContextClassLoader().loadClass(path));
+                            } catch (ClassNotFoundException e) {
+                                // log.error("添加用戶自定義視圖類錯誤找不到此類的.class文件");
+                                System.out.println("添加用戶自定義視圖類錯誤找不到此類的.class文件");
```

Comment:
Remove Chinese comments and add necessary comments in English.

If you are able to follow clean code, you don't need comments here, since you can extract some of them to the new methods.